wallet/walletrpc.c

```diff
@@ -1275,13 +1276,12 @@ static struct command_result *json_reserveinputs(struct command *cmd,
 	 * around, so we remove the auto-cleanup that happens
 	 * when the utxo objects are free'd */
 	wallet_persist_utxo_reservation(cmd->ld->wallet, utx->wtx->utxos);
-	/* Set the lease on these utxos to expire in ~24 hours,
-	 * or 24 blocks */
-	u32 expires_at = cmd->ld->topology->tip->height + 6 * 24;
-	for (size_t i = 0; i < tal_count(utx->wtx->utxos); i++)
-		wallet_output_reservation_update(cmd->ld->wallet,
-						 utx->wtx->utxos[i],
-						 expires_at);
+	if (*expire_at > 0) {
+		for (size_t i = 0; i < tal_count(utx->wtx->utxos); i++)
+			wallet_output_reservation_update(cmd->ld->wallet,
+							 utx->wtx->utxos[i],
+							 *expire_at);
```

Comment:
Seems to me like this uses the relative expiry as absolute blockheight. Should probably be something like the following:

```suggestion
		for (size_t i = 0; i < tal_count(utx->wtx->utxos); i++)
			wallet_output_reservation_update(cmd->ld->wallet,
							 utx->wtx->utxos[i],
							 cmd->ld->topology->tip->height + *expire_at);
```

Reply:
Ah yes, good spot.